api/src/lib/webhonc/index.ts

```diff
@@ -0,0 +1,161 @@
+import path from "node:path";
+import type { LibSQLDatabase } from "drizzle-orm/libsql";
+import WebSocket from "ws";
+import type { z } from "zod";
+import * as schema from "../../db/schema.js";
+import logger from "../../logger.js";
+import { resolveServiceArg } from "../../probe-routes.js";
+import {
+  executeProxyRequest,
+  handleFailedRequest,
+  handleSuccessfulRequest,
+} from "../proxy-request/index.js";
+import { WsMessageSchema } from "../types.js";
+import { resolveUrlQueryParams } from "../utils.js";
+import { getWebHoncConnectionId, setWebHoncConnectionId } from "./store.js";
+
+export function connectToWebhonc(
+  host: string,
+  db: LibSQLDatabase<typeof schema>,
+  wsConnections: Set<WebSocket>,
+) {
+  const protocol = host.startsWith("localhost") ? "ws" : "wss";
+  const socket = new WebSocket(`${protocol}://${host}/ws`);
+
+  socket.onopen = () => {
+    logger.debug(`Connected to the webhonc service at ${host}`);
+  };
+
+  socket.onmessage = async (event) => {
+    logger.debug("Received message from the webhonc service:", event.data);
+    await handleMessage(event, wsConnections, db);
+  };
+
+  socket.onclose = (event) => {
```

Comment:
for the future: could be good to maintain some state about the webhonc connection and propagate that to the UI somehow

Reply:
yea that's a good point, adding a `TODO:` comment for now